backend/pick-git/src/test/java/com/woowacourse/pickgit/tag/application/TagServiceIntegrationTest.java

```diff
@@ -0,0 +1,105 @@
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.web.client.HttpClientErrorException;
+
+@DataJpaTest
+class TagServiceIntegrationTest {
+
+    private TagService tagService;
+
+    @Autowired
+    private TagRepository tagRepository;
+
+    private String accessToken = "oauth.access.token";
+    private String userName = "jipark3";
+    private String repositoryName = "doms-react";
+    private ObjectMapper objectMapper = new ObjectMapper();
+
+    @BeforeEach
+    void setUp() {
+        PlatformTagExtractor platformTagExtractor =
+            new GithubTagExtractor(new MockTagApiRequester(), objectMapper);
+        tagService = new TagService(platformTagExtractor, tagRepository);
+    }
+
+    @DisplayName("Repository에 포함된 언어 태그를 추출한다.")
+    @Test
+    void extractTags_ValidRepository_ExtractionSuccess() {
+        ExtractionRequestDto extractionRequestDto =
+            new ExtractionRequestDto(accessToken, userName, repositoryName);
+        List<String> tags = Arrays.asList("JavaScript", "HTML", "CSS");
+
+        TagsDto tagsDto = tagService.extractTags(extractionRequestDto);
+
+        assertThat(tagsDto.getTags()).containsAll(tags);
+    }
+
+    @DisplayName("잘못된 경로로 태그 추출 요청시 404 예외가 발생한다.")
+    @Test
+    void extractTags_InvalidUrl_ExceptionThrown() {
+        String userName = "nonuser";
+        String repositoryName = "nonrepo";
+        ExtractionRequestDto extractionRequestDto =
+            new ExtractionRequestDto(accessToken, userName, repositoryName);
+
+        assertThatCode(() -> tagService.extractTags(extractionRequestDto))
+            .isInstanceOf(HttpClientErrorException.class)
+            .hasMessageContaining("404");
+    }
+
+    @DisplayName("유효하지 않은 토큰으로 태그 추출 요청시 401 예외가 발생한다.")
+    @Test
+    void extractTags_InvalidToken_ExceptionThrown() {
+        String accessToken = "invalidtoken";
+        ExtractionRequestDto extractionRequestDto =
+            new ExtractionRequestDto(accessToken, userName, repositoryName);
+
+        assertThatCode(() -> tagService.extractTags(extractionRequestDto))
+            .isInstanceOf(HttpClientErrorException.class)
+            .hasMessageContaining("401");
+    }
+
+    @DisplayName("태그 이름을 태그로 변환한다.")
+    @Test
+    void findOrCreateTags_ValidTag_TransformationSuccess() {
+        tagRepository.save(new Tag("tag3"));
+        List<String> tagNames = Arrays.asList("tag1", "tag2", "tag3");
+        TagsDto tagsDto = new TagsDto(tagNames);
+
+        List<String> tags = tagService.findOrCreateTags(tagsDto)
+            .stream()
+            .map(Tag::getName)
+            .collect(Collectors.toList());
```

Comment:
```suggestion
            .collect(toList());
```

여기도 static import 해주세요 !